scripts/builtin/xgboost.dml

```diff
@@ -0,0 +1,780 @@
+                else if(as.scalar(curr_value) < 1000)
+                    print_string = print_string + toString(curr_value, linesep="   ", sep=",")
+                else if(as.scalar(curr_value) < 10000)
+                    print_string = print_string + toString(curr_value, linesep="  ", sep=",")
+                else
+                    print_string = print_string + toString(curr_value, linesep=" ", sep=",")
+            }
+            print(print_string)
+        }
+    }
+    print("--------------------------------------")
+}
+
+
+printMMatrix = function(Matrix[Double] m) return() {
+    print("--------------------------------------")
+    print(":: [MATRIX] OUTPUT M MATRIX  (col:" + ncol(m) + ",row:" + nrow(m) + ")")
+
+    if(nrow(m) != 0 & ncol(m) != 0)
+    {
+        col_string = "::::::::::::::::::::::::"
+        for(col in 1:ncol(m)) {
+            if(col < 10)
+                col_string = col_string + "::(" + toString(col) + "):::"
+            else
+                col_string = col_string + "::(" + toString(col) + ")::"
+        }
+        print(col_string)
+        for(row in 1:nrow(m)) {
+
+            if(row == 1)
+                print_string = "             node id (" + row + ")  "
+            if(row == 2)
+                print_string = "             tree id (" + row + ")  "
+            if(row == 3)
+                print_string = "offset to left child (" + row + ")  "
+            if(row == 4)
+                print_string = "used feature col idx (" + row + ")  "
+            if(row == 5)
+                print_string = " leaf=0/scal=1/cat=2 (" + row + ")  "
+            if(row == 6)
+                print_string = "   scalar=splitvalue (" + row + ")  "
+
+            for(col in 1:ncol(m)) {
+                curr_value = m[row,col]
+                if(as.scalar(curr_value) == 'NaN')
+                    print_string = print_string + toString(curr_value, linesep="     ", sep=",")
+                else if(as.scalar(curr_value) < 0)
+                    print_string = print_string + toString(curr_value, linesep="  ", sep=",")
+                else if(as.scalar(curr_value) < 10)
+                    print_string = print_string + toString(curr_value, linesep="   ", sep=",")
+                else if(as.scalar(curr_value) < 100)
+                    print_string = print_string + toString(curr_value, linesep="  ", sep=",")
+                else if(as.scalar(curr_value) < 1000)
+                    print_string = print_string + toString(curr_value, linesep=" ", sep=",")
+                else if(as.scalar(curr_value) < 10000)
+                    print_string = print_string + toString(curr_value, linesep=" ", sep=",")
+                else
+                    print_string = print_string + toString(curr_value, linesep=" ", sep=",")
+            }
+            print(print_string)
+        }
+    }
+    print("--------------------------------------")
+}
```

Comment:
remember a newline in the end of the script to make GitHub happy.

Reply:
done